### PR TITLE
Display SubTypeAccordions

### PR DESCRIPTION
**Pull request by gilhsu**
- modify data to inject `subType` as top-level key
- parse `resourceIdsGroupedByType` state to included `subType`. Now: `category > subType > set(resourceIds)`
- build `SubTypeAccordion` via `native-base`
- connect `CategorySelector` to display the correct `SubTypeAccordion`s

image

### CODE REVIEW

File: src/resources/fhirReader.js [hunk 1]
```diff
@@ -2,6 +2,35 @@ import {
...
+      subType = resource.vaccineCode?.text;
+      break;
+    case 'MedicationRequest':
+      subType = resource.medicationCodeableConcept?.text;
+      break;
+    case 'CarePlan':
+      subType = resource.category?.[0]?.text;
+      break;
+    default:
+      subType = null;
```
Comment (vadi2) [inline src/resources/fhirReader.js:27]: A warning here so it's not a bit bucket?

**Review by vadi2 — APPROVED** [review]
It works :+1:

File: src/resources/fhirReader.js (hunk 1 above)
Comment (foxtree) [reply src/resources/fhirReader.js:27]: yes: a warning for unsupported types.

and, possibly: make the subType `"Other"`?

File: src/components/CategorySelector/CategorySelector.js [hunk 2]
```diff
@@ -1,48 +1,74 @@
...
+    );
+  };
+
+  CategoryButton.propTypes = {
+    category: string.isRequired,
+  };
+
+  return (
+    <ScrollView style={styles.root} horizontal showsHorizontalScrollIndicator={false}>
+      {categories.map((category) => {
```
Comment (foxtree) [inline src/components/CategorySelector/CategorySelector.js:29]: This should just use `supportedResourcesSelector` from selectors -- it is already filtering-out invalid categories by `RESOURCE_TYPES`, and correctly sorting by the `RESOURCE_TYPES` label.

...`supportedResourcesSelector` can also adapted for filtering, or replaced with a new selector that building on `supportedResourcesSelector` as a memoized input.

File: src/components/CategorySelector/CategorySelector.js [hunk 3]
```diff
@@ -1,48 +1,74 @@
...
+        if (category !== 'Patient') {
+          return <CategoryButton key={category} category={category} />;
```
Comment (foxtree) [inline src/components/CategorySelector/CategorySelector.js:31]: see `src/screens/SummaryScreen.js` for how to `mapDispatchToProps`, and:

```
          {Object.entries(resourceIdsGroupedByType).map(
            ([resourceType, resourceIds]) => (
              <ResourceTypeRow
                key={resourceType}
                resourceIds={resourceIds} // or, use additional selector to derive underlying resources or groups of resources
              />
            ),
          )}
```

File: src/components/CategorySelector/CategorySelector.js (hunk 2 above)
Comment (gilhsu) [reply src/components/CategorySelector/CategorySelector.js:29]: This is applied in PR#29. This PR is really just about `SubTypeAccordions` working.

File: src/screens/CatalogScreen.js [hunk 4]
```diff
@@ -1,28 +1,52 @@
...
-        <TimelineWidget />
-        <CategorySelector />
-        <RecordCardsContainer />
-      </ScrollView>
-    </FilterDrawer>
-  </SafeAreaView>
-);
+const CatalogScreen = ({ resourceIdsGroupedByType }) => {
+  const categories = Object.keys(resourceIdsGroupedByType);
+  const scrubbedCategories = categories.filter((category) => category !== 'Patient' && category !== 'Observation');
```
Comment (foxtree) [inline src/screens/CatalogScreen.js:16]: `resourceIdsGroupedByType`, from the selector, is already filtered by valid categories (and omitting Patient)

File: src/screens/CatalogScreen.js [hunk 5]
```diff
@@ -1,28 +1,52 @@
...
+  const [selectedCategory, setSelectedCategory] = useState(scrubbedCategories[0]);
```
Comment (foxtree) [inline src/screens/CatalogScreen.js:17]: this should be derived via a selector (either existing, or new), and `mapStateToProps`

File: src/screens/CatalogScreen.js [hunk 6]
```diff
@@ -1,28 +1,52 @@
...
+    </SafeAreaView>
+  );
+};
+
+CatalogScreen.propTypes = {
+  resourceIdsGroupedByType: shape({}).isRequired,
+};
+
+const mapStateToProps = (state) => ({
+  resourceIdsGroupedByType: state.resourceIdsGroupedByType,
```
Comment (foxtree) [inline src/screens/CatalogScreen.js:46]: See: `src/screens/SummaryScreen.js`: 
This should be using the existing `supportedResourcesSelector(state)`  (which already filters supported categories, _and sorts_ them by their label)

...if it lacks the necessary structure substructure, a new selector can be created, derived from `supportedResourcesSelector`

File: src/components/CategorySelector/CategorySelector.js [hunk 7]
```diff
@@ -1,48 +1,74 @@
...
+  };
+
+  CategoryButton.propTypes = {
+    category: string.isRequired,
+  };
+
+  return (
+    <ScrollView style={styles.root} horizontal showsHorizontalScrollIndicator={false}>
+      {categories.map((category) => {
+        if (category !== 'Patient') {
```
Comment (foxtree) [inline src/components/CategorySelector/CategorySelector.js:30]: this check should be unnecessary, if `categories` prop was already filtered by `RESOURCE_TYPES`
(e.g.: using `supportedResourcesSelector`, or new selector that is derived from `supportedResourcesSelector`)

File: src/screens/CatalogScreen.js (hunk 4 above)
Comment (gilhsu) [reply src/screens/CatalogScreen.js:16]: this is addressed in #29

File: src/screens/CatalogScreen.js (hunk 5 above)
Comment (gilhsu) [reply src/screens/CatalogScreen.js:17]: this is addressed in #29

File: src/screens/CatalogScreen.js (hunk 6 above)
Comment (gilhsu) [reply src/screens/CatalogScreen.js:46]: this is addressed in #29

**Pull request closed without merging**